File: SportsStore.Domain/Entities/OrderStatus.cs
namespace SportsStore.Domain.Entities;

public enum OrderStatus
{
    Submitted = 0,
    InventoryPending = 1,
    InventoryConfirmed = 2,
    InventoryFailed = 3,
    PaymentPending = 4,
    PaymentApproved = 5,
    PaymentFailed = 6,
    ShippingPending = 7,
    ShippingCreated = 8,
    Completed = 9,
    Failed = 10
}

File: admin-dashboard/src/orderStatus.ts
export const ORDER_STATUSES = [
  'Submitted',
  'InventoryPending',
  'InventoryConfirmed',
  'InventoryFailed',
  'PaymentPending',
  'PaymentApproved',
  'PaymentFailed',
  'ShippingPending',
  'ShippingCreated',
  'Completed',
  'Failed',
] as const

export type OrderStatusValue = (typeof ORDER_STATUSES)[number]

const FAILURE_STATUS_LOOKUP = new Set<OrderStatusValue>(['InventoryFailed', 'PaymentFailed', 'Failed'])

export function isFailureStatus(status: string): boolean {
  return FAILURE_STATUS_LOOKUP.has(status as OrderStatusValue)
}
